### PR TITLE
Fixes #19417 - Safely check params for nested keys

### PR DESCRIPTION
**Pull request by tbrisker**

### CODE REVIEW

File: app/controllers/concerns/foreman/controller/parameters/keep_param.rb [hunk 1]
```diff
@@ -8,7 +8,7 @@ module Foreman::Controller::Parameters::KeepParam
   def keep_param(params, top_level_hash, *keys)
     # Delete keys being kept from the `params` hash, so the block yielded to filters the others
     old_params = keys.inject({}) do |op,(key,val)|
-      if params[top_level_hash].has_key?(key)
+      if params[top_level_hash].try(:has_key?, key)
```
Comment (domcleal) [inline app/controllers/concerns/foreman/controller/parameters/keep_param.rb]: Nitpick, prefer `try!`.

**Review by domcleal — CHANGES_REQUESTED** [review]
Thanks, could you please add a test to `test/unit/concerns/parameters/keep_param_test.rb`?

**Comment by tbrisker** [comment]
@domcleal - Fixed and added test.

**Review by domcleal — APPROVED** [review]

**Comment by domcleal** [comment]
Merged, thanks @tbrisker.

**Pull request merged**